src/ReactiveUI/ViewLocator.cs

```diff
@@ -93,31 +95,27 @@ public DefaultViewLocator(Func<string, string> viewModelToViewFunc = null)
         public IViewFor ResolveView<T>(T viewModel, string contract = null)
             where T : class
         {
-            var view = this.AttemptViewResolutionFor(viewModel.GetType(), contract);
+            var view = GetCandidateViewModelTypes(viewModel)
+                .Select(type => AttemptViewResolutionFor(type, contract))
+                .FirstOrDefault(x => x != null);
 
-            if (view != null) {
-                return view;
-            }
-
-            view = this.AttemptViewResolutionFor(typeof(T), contract);
-
-            if (view != null) {
-                return view;
-            }
-
-            var toggledType = ToggleViewModelType(viewModel);
-            view = this.AttemptViewResolutionFor(toggledType, contract);
-
-            if (view != null) {
-                return view;
-            }
+            if (view != null) return view;
 
             this.Log().Warn("Failed to resolve view for view model type '{0}'.", typeof(T).FullName);
             return null;
         }
 
+        private IEnumerable<Type> GetCandidateViewModelTypes<T>(T viewModel)
+        {
+            yield return viewModel.GetType();
+            yield return typeof(T);
+            yield return ToggleViewModelType(viewModel.GetType());
+            yield return ToggleViewModelType(typeof(T));
+        }
+
         private IViewFor AttemptViewResolutionFor(Type viewModelType, string contract)
         {
+            if (viewModelType == null) return null;
```

Comment:
This short circuit handles #1298 we could instead handle the null better out of ToggleViewModelType, but I think it's a reasonable thing to have.